h5grove/utils.py

```diff
@@ -212,7 +212,7 @@ def _sanitize_dtype(dtype: np.dtype) -> np.dtype:
 
     :raises ValueError: If trying to sanitize a non-numeric numpy dtype
     """
-    if dtype.kind not in ("f", "i", "u"):
```

Comment:
Add a little comment that explains why `b` is actually numeric ?

Reply:
Well, `b` means `byte`, not boolean, so it was more of an oversight.

Now that you mention it, I feel like this check doesn't really make sense. The purpose of `_sanitize_dtype` is to make specific dtypes safe:

- big endian to little endian
- float16 to float32
- float128 to float64

It shouldn't care about anything else. The `convert` function, however, does check that `dtype=safe` is only used with numeric datasets as you would expect. It does so by calling `is_numeric_data`, which does check for booleans already.

I'll push a separate commit with a proposed refactoring.